client/extensions/zoninator/state/locks/reducer.js

```diff
@@ -0,0 +1,67 @@
+/** @format */
+
+/**
+ * Internal dependencies
+ */
+import { combineReducers, createReducer, keyedReducer } from 'state/utils';
+import {
+	ZONINATOR_REQUEST_LOCK,
+	ZONINATOR_REQUEST_LOCK_ERROR,
+	ZONINATOR_UPDATE_LOCK,
+} from '../action-types';
+
+const isRequesting = createReducer(
+	{},
+	{
+		[ ZONINATOR_REQUEST_LOCK ]: () => true,
+		[ ZONINATOR_REQUEST_LOCK_ERROR ]: () => false,
+		[ ZONINATOR_UPDATE_LOCK ]: () => false,
+	}
+);
+
+export const requesting = keyedReducer( 'siteId', keyedReducer( 'zoneId', isRequesting ) );
+
+const isBlocked = createReducer(
+	{},
+	{
+		[ ZONINATOR_UPDATE_LOCK ]: () => false,
+		[ ZONINATOR_REQUEST_LOCK_ERROR ]: ( state, { blocked } ) => blocked,
+	}
+);
+
+export const blocked = keyedReducer( 'siteId', keyedReducer( 'zoneId', isBlocked ) );
```

Comment:
Maybe consider inlining `isRequesting` and `isBlocked` into `requesting` and `blocked`, respectively? Using those slight name variations to denote something rather specific (basic vs keyed reducer variations) can be a bit misleading...

Reply:
Also, shouldn't this reducer be called `locked` instead of `blocked`? 😄

Reply:
I called it `blocked` specifically so it clearly indicates the zone is being locked by another user. Using `locked` might suggest it's locked by the current user.